core/src/main/java/org/infinispan/marshall/core/internal/StreamBytesObjectOutput.java

```diff
@@ -0,0 +1,103 @@
+package org.infinispan.marshall.core.internal;
+
+import java.io.IOException;
+import java.io.ObjectOutput;
+import java.io.OutputStream;
+
+final class StreamBytesObjectOutput implements ObjectOutput {
```

Comment:
Not always. JGroups' RpcDispatcher.Marshaller implementation calls objectToBuffer which does not any such wrapping.